bundles/org.openhab.binding.mielecloud/src/main/java/org/openhab/binding/mielecloud/internal/config/servlet/CreateBridgeServlet.java

```diff
@@ -127,10 +127,11 @@ private Thing pairOrReconfigureBridge(String locale, ThingUID bridgeUid, String
                 .withProperty(Thing.PROPERTY_MODEL_ID, MIELE_CLOUD_BRIDGE_NAME)
                 .withProperty(MieleCloudBindingConstants.CONFIG_PARAM_LOCALE, locale)
                 .withProperty(MieleCloudBindingConstants.CONFIG_PARAM_EMAIL, email).build();
-        if (inbox.add(result)) {
-            return pairBridge(bridgeUid);
-        } else {
+        if (thingRegistry.get(bridgeUid) != null) {
             return reconfigureBridge(bridgeUid);
+        } else {
+            inbox.add(result);
```

Comment:
Wouldn't `inbox.approve` in `pairBridge` fail if the discovery result cannot be added instantly? Shouldn't the result of the future be awaited here and the pairing only proceed if the discovery result was successfully added to the inbox? Awaiting should be done with a reasonable timeout because the user is waiting for the success / failure page to load in the browser.

Reply:
Probably. Can you prepare a PR for that?

Reply:
Yes.